polus-generalized-linear-model-plugin/src/main.R

```diff
@@ -0,0 +1,226 @@
+  invisible(NULL)
+}
+addModelArgs <- function(parser) {
+  parser$add_argument("--modeltype",type = "character",help="Select either binomial or gaussian or Gamma or poisson or quasi or quasibinomial or quasipoisson or negativebinomial")
+  invisible(NULL)
+}
+addOutputArgs <- function(parser) {
+  parser$add_argument("--outdir", type = "character",help="Output csv file")
+  invisible(NULL)
+}
+getAllParser <- function() {
+  parser <- ArgumentParser(description="ALL PARSER")
+  addRequiredArgs(parser)
+  addPredictArgs(parser)
+  addExcludeArgs(parser)
+  addMethodArgs(parser)
+  addModelArgs(parser)
+  addOutputArgs(parser)
+  return(parser)
+}
+
+# Parse the arguments
+parser <- getAllParser()
+args <- parser$parse_args()
+
+#Path to csvfile directory
+inpfile <- args$inpdir
+loginfo('inpfile = %s', inpfile)
+
+#Column to be predicted
+predictcolumn <- args$predictcolumn
+loginfo('predictcolumn = %s', predictcolumn)
+
+#Columns to be excluded
+exclude_col <- args$exclude
+excludes<-as.list(strsplit(exclude_col, ",")[[1]])
+loginfo('exclude = %s', excludes)
+
+#Select glmmethod-primary or interaction or secondorder
+glmmethod <- args$glmmethod
+loginfo('glmmethod = %s', glmmethod)
+
+#Select modeltype based on distribution of data
+modeltype <- args$modeltype
+loginfo('modeltype = %s', modeltype)
+
+#Path to save output csvfiles
+csvfile <- args$outdir
+loginfo('csvfile = %s', csvfile)
+
+#Get list of .csv files in the directory including sub folders for modeling
+files_to_read = list.files(
+  path = inpfile,        
+  pattern = ".*csv", 
+  recursive = TRUE,          
+  full.names = TRUE
+)
+
+#Read the csv files
+datalist = lapply(files_to_read,read.csv)
+
+for (dataset in datalist) {
+  for (file_csv in files_to_read) {
+    #Get filename
+    file_name <- SplitPath(file_csv)$filename
+    
+    #Check whether any column needs to be excluded
+    if(length(excludes) > 0) {
+      for (i in 1:length(excludes)) {
+        if(!excludes[i] %in% colnames(dataset)) {
+          logwarn('column to exclude from %s is not found',file_name)
+        }
+      }
+      datasub <-dataset[ , !(names(dataset) %in% excludes)]
+    }
+    else if(length(excludes) == 0) {
+      datasub <-dataset
+    }
+    # Remove columns with all values as zero
+    datasub <- datasub[colSums(datasub) > 0]
+    
+    #Check whether predict column is present in dataframe
+    if(!(predictcolumn %in% colnames(datasub))) {
+      logwarn('predict column name is not found in %s',file_name)
+      next
+    }
+    
+    #Get column names without predict variable
+    drop_dep <- datasub[ , !(names(datasub) %in% predictcolumn)]
+    resp_var <- colnames(drop_dep)
+    
+    if((modeltype == 'Gaussian') || (modeltype == 'Poisson') || (modeltype == 'Binomial') || (modeltype == 'Multinomial') || (modeltype == 'Quasibinomial') || (modeltype == 'Quasipoisson') || (modeltype == 'Quasi')) {
```

Comment:
This should be made easier to read. Maybe make the possible options a list, and check to see if the value is in the list.